Wikipedia/Code/WMFSearchViewController.m

```diff
@@ -364,16 +368,19 @@ - (BOOL)textFieldShouldClear:(UITextField *)textField {
 
 #pragma mark - Search
 
-- (void)setSearchTerm:(NSString *)searchTerm {
-    if (searchTerm.length == 0) {
+- (void)performSearchWithCurrentSearchTerm {
+    if (!self.viewIfLoaded) {
         return;
     }
-    [self setSearchFieldText:searchTerm];
-    [self searchForSearchTerm:searchTerm];
+    if (self.searchTerm.length == 0) {
```

Comment:
In the old version, there wasn't a trim here. This line was just moved down, not added.